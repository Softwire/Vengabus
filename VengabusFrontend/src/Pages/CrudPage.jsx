import React, { Component } from 'react';
import { CrudInterface } from '../Components/CrudInterface';

export class CrudPage extends Component {

    render() {
        return (
            <CrudInterface {...this.props} />
        );
    }
}
