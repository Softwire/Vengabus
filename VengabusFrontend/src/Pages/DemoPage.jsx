import React, { Component } from 'react';
import { QueueList } from '../Components/QueueList';
import { MessageList } from '../Components/MessageList';
import { css } from 'react-emotion';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { Glyphicon, Button } from 'react-bootstrap';
import { PAGES, pageSwitcher } from './PageSwitcherService';

export class DemoPage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            retrievedData: [{ number: 1, name: 'q1', status: 'active' }, { number: 2, name: 'q2', status: 'active' }, { number: 3, name: 'q3', status: 'dead' }],
            messageData: [
                { predefinedProperties: { messageId: "mal formatted xml 1" }, messageBody: "<a><shipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>", type: "XML" },
                { predefinedProperties: { messageId: "perfect XML" }, messageBody: "<a><b>sdfds</b><c>sdgsdg</c></a>" },
                { predefinedProperties: { messageId: "mal formatted xml 2" }, messageBody: "<a>{s}<hipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>", type: "XML" },
                { predefinedProperties: { messageId: "Error causing xml" }, messageBody: "<ashipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>", type: "XML" },
                { predefinedProperties: { messageId: "Error causing xml 2" }, messageBody: "<shipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder>", type: "XML" },
                { predefinedProperties: { messageId: "good json" }, messageBody: `{"result":true , "count":42}`, },
                { predefinedProperties: { messageId: "bracket json" }, messageBody: `[true , 42]`, },
                { predefinedProperties: { messageId: "one long line" }, messageBody: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`, },
                { predefinedProperties: { messageId: "one short line" }, messageBody: `Lorem ipsum dolor sit amet`, },
                { predefinedProperties: { messageId: "wide xml" }, messageBody: `<someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping>`, },
                { predefinedProperties: { messageId: "mal formatted wide xml" }, messageBody: `<someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping><someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping></someModeratelyLongXMLtagForTheDemoPageToTestWordWrapping>`, },
                { predefinedProperties: { messageId: "wide json" }, messageBody: `{"result":true , "some numerical array":[234324,457,67869867,345435,324234,45765765,78756,435,2345]}`, },
                { predefinedProperties: { messageId: "mal formatted wide json" }, messageBody: `{"result":true , "some numerical array":[234324,457,67869867,345435,324234,45765765,78756,435,2345]]}`, },
                {
                    predefinedProperties: { messageId: "text with newline characters" },
                    messageBody:
                        `Lorem ipsum dolor 
    sit amet Lorem ipsum dolor 
sit amet \nthat was a newline`,
                },
            ]
        };
    }

    updateRetrievedData = (data) => {
        this.setState({ retrievedData: data });
    };


    handleListQueueMessages = (queueName) => {
        const serviceBusService = serviceBusConnection.getServiceBusService();
        const messagePromise = serviceBusService.listQueueMessages(queueName);
        messagePromise.then((result) => console.log(result.data));
    }

    handleListSubscriptionMessages = (topicName, subscriptionName) => {
        const serviceBusService = serviceBusConnection.getServiceBusService();
        const messagePromise = serviceBusService.listSubscriptionMessages(topicName, subscriptionName);
        messagePromise.then((result) => console.log(result.data));
    }


    render() {
        const queueDivStyle = css`
            width: 30%;
            margin: 10px;
            display: inline-block; /*to allow tables to be displayed side by side*/
        `;
        // Hard coded message to replay for now
        const exampleMessage = {
            customProperties: {
                userDefinedProp1: 'value1',
                userDefinedProp2: 'value2'
            },
            predefinedProperties: {
                MessageId: 'Message1',
                ContentType: 'null'
            },
            MessageBody: 'Hello world!'
        };

        //qq remove hardcoded endpoint names later
        const queueName = 'demoqueue1';
        const topicName = 'demotopic1';
        const subscriptionName = 'demosubscription1';
        const serviceBusService = serviceBusConnection.getServiceBusService();

        return (
            < div >
                <div className={queueDivStyle}>
                    <button onClick={() => pageSwitcher.switchToPage(PAGES.SendMessagePage, exampleMessage)}>Replay Example Message</button>
                    <QueueList queueData={this.state.retrievedData} />
                    <MessageList messageData={this.state.messageData} />
                    {/*qq delete the text in Button once implemented properly*/}
                    <Button onClick={() => serviceBusService.deleteQueueMessages(queueName)} bsStyle="danger">
                        Delete queue messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>
                    <Button onClick={() => serviceBusService.deleteTopicMessages(topicName)} bsStyle="danger">
                        Delete topic messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>
                    <Button onClick={() => serviceBusService.deleteSubscriptionMessages(topicName, subscriptionName)} bsStyle="danger">
                        Delete subcription messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>

                </div>

                <div>
                    <p>Displays other Page</p>
                    <Button onClick={() => this.handleListQueueMessages(queueName)}>
                        Display {queueName} messages
                </Button>
                    <Button onClick={() => this.handleListSubscriptionMessages(topicName, subscriptionName)}>
                        Display {subscriptionName} messages in topic {topicName}
                    </Button>
                </div>
            </div >
        );
    }
}
