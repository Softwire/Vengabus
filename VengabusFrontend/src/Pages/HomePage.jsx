import React, { Component } from 'react';
import { QueueList } from '../Components/QueueList';
import { MessageList } from '../Components/MessageList';
import { css } from 'react-emotion';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { Glyphicon, Button } from 'react-bootstrap';
import { PAGES, pageSwitcher } from './PageSwitcherService';

export class HomePage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            retrievedData: undefined,
            messageData: [
                { messageId: 10, messageBody: "<a><shipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>" ,type : "XML"},
                { messageId: 11, messageBody: "<a><b>sdfds</b><c>sdgsdg</c></a>" },
                { messageId: 12, messageBody: "<a>{s}<hipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>",type: "XML" },
                { messageId: 13, messageBody: "<ashipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder></a>", type: "XML"},
                { messageId: 14, messageBody: "<shipto><name>Ola Nordmann</name><address>Langgt 23</address><city>4000 Stavanger</city><country>Norway</country></shipto><shiporder><item><title>Empire Burlesque</title><note>Special Edition</note><quantity>1</quantity><price>10.90</price></item><item><title>Hide your heart</title><quantity>1</quantity><price>9.90</price></item></shiporder>",type: "XML" },
                { messageId: 15, messageBody: `{"result":true , "count":42}` ,},
                { messageId: 15, messageBody: `apple` ,},
            ]
        };
    }

    updateRetrievedData = (data) => {
        this.setState({ retrievedData: data });
    };


    render() {
        const queueDivStyle = css`
            width: 30%;
            margin: 10px;
            display: inline-block; /*to allow tables to be displayed side by side*/
        `;
        // Hard coded message to replay for now
        const exampleMessage = {
            MessageProperties: {
                userDefinedProp1: 'value1',
                userDefinedProp2: 'value2'
            },
            MessageBody: 'Hello world!',
            MessageId: 'Message1',
            ContentType: 'null'
        };

        //qq remove hardcoded endpoint names later
        const queueName = 'demoqueue1';
        const topicName = 'demotopic1';
        const subscriptionName = 'demosubscription1';
        const serviceBusService = serviceBusConnection.getServiceBusService();

        return (
            < div >
                <div className={queueDivStyle}>
                    <button onClick={() => pageSwitcher.switchToPage(PAGES.SendMessagePage, exampleMessage)}>Replay Example Message</button>
                    <QueueList queueData={this.state.retrievedData} />
                    <MessageList messageData={this.state.messageData} />
                    {/*qq delete the text in Button once implemented properly*/}
                    <Button onClick={() => serviceBusService.deleteQueueMessages(queueName)} bsStyle="danger">
                        Delete queue messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>
                    <Button onClick={() => serviceBusService.deleteTopicMessages(topicName)} bsStyle="danger">
                        Delete topic messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>
                    <Button onClick={() => serviceBusService.deleteSubscriptionMessages(topicName, subscriptionName)} bsStyle="danger">
                        Delete subcription messages&#160;
                        <Glyphicon glyph="trash" />
                    </Button>

                </div>
            </div >
        );
    }
}
