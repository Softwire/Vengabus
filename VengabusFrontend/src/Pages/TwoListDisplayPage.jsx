import React, { Component } from 'react';
import { serviceBusConnection } from '../AzureWrappers/ServiceBusConnection';
import { QueueList } from '../Components/QueueList';
import { TopicList } from '../Components/TopicList';
import { MessageList } from '../Components/MessageList';
import { css, injectGlobal } from 'react-emotion';
import { Button, Breadcrumb } from 'react-bootstrap';
import { SubscriptionList } from '../Components/SubscriptionList';
import { EndpointTypes, typeToTitle } from '../Helpers/EndpointTypes';


export class TwoListDisplayPage extends Component {
    constructor(props) {
        super(props);
        this.displayHistory = [];
        this.breadCrumbHistory = [{ name: "Home", type: undefined }];
        this.state = {
            queueData: undefined,
            topicData: undefined,
            subscriptionData: undefined,
            messageData: undefined,
            rightTableType: EndpointTypes.TOPIC
        };
    }

    componentDidMount() {
        serviceBusConnection.registerForUpdatesPrompts(this.resetInitialStateData);

    }
    componentWillUnmount() {
        serviceBusConnection.deregisterForUpdatesPrompts(this.resetInitialStateData);

    }

    handleQueueRowClick = (e, row, rowIndex) => {
        this.breadCrumbHistory = [{ name: "Home", type: undefined }, { name: row.name, type: EndpointTypes.QUEUE }];
        this.updateEndpointMessageData();
        this.setState({
            rightTableType: EndpointTypes.MESSAGE
        });
    }

    handleTopicRowClick = (e, row, rowIndex) => {
        this.breadCrumbHistory = [{ name: "Home", type: undefined }, { name: row.name, type: EndpointTypes.TOPIC }];
        this.updateTopicSubscriptionData();
        this.setState({
            rightTableType: EndpointTypes.SUBSCRIPTION
        });
    }

    handleSubscriptionRowClick = (e, row, rowIndex) => {
        this.breadCrumbHistory[2] = { name: row.name, type: EndpointTypes.SUBSCRIPTION };
        this.updateEndpointMessageData();
        this.setState({
            rightTableType: EndpointTypes.MESSAGE
        });
    }


    updateAllQueueData = () => {
        const serviceBusService = serviceBusConnection.getServiceBusService();
        const fetchedQueueData = serviceBusService.listQueues();
        fetchedQueueData.then(result => {
            this.setState({
                queueData: result.data
            });
        }
        );
    }

    updateAllTopicData = () => {
        const serviceBusService = serviceBusConnection.getServiceBusService();
        const fetchedTopicData = serviceBusService.listTopics();
        fetchedTopicData.then(result =>
            this.setState({
                topicData: result.data
            })
        );
    }

    updateTopicSubscriptionData = () => {
        const topicName = this.breadCrumbHistory[this.breadCrumbHistory.length - 1].name;
        const serviceBusService = serviceBusConnection.getServiceBusService();
        const fetchedSubscriptionData = serviceBusService.listSubscriptions(topicName);
        fetchedSubscriptionData.then(result => {
            this.setState({
                subscriptionData: result.data
            })
        }
        );
    }

    updateEndpointMessageData = () => {
        const serviceBusService = serviceBusConnection.getServiceBusService();
        let fetchedMessageData;
        const leftTableType = this.breadCrumbHistory[this.breadCrumbHistory.length - 1].type;
        if (leftTableType === EndpointTypes.QUEUE || leftTableType === undefined) {
            const queueName = this.breadCrumbHistory[this.breadCrumbHistory.length - 1].name;
            fetchedMessageData = serviceBusService.listQueueMessages(queueName);
        } else {
            const topicName = this.breadCrumbHistory[this.breadCrumbHistory.length - 2].name;
            const subscriptionName = this.breadCrumbHistory[this.breadCrumbHistory.length - 1].name;
            fetchedMessageData = serviceBusService.listSubscriptionMessages(topicName, subscriptionName);
        }
        fetchedMessageData.then((result) => {
            this.setState({
                messageData: result.data
            });
        });
    }

    resetInitialStateData = () => {
        this.updateAllTopicData();
        this.updateAllQueueData();
        this.breadCrumbHistory = [{ name: "Home", type: undefined }];
        this.setState({
            rightTableType: EndpointTypes.TOPIC
        });
    };

    getList = (isForRightHandList) => {
        let typeOfData;
        const currentLeftTable = this.breadCrumbHistory[this.breadCrumbHistory.length - 1];
        if (isForRightHandList) {
            typeOfData = this.state.rightTableType;
        } else {
            //if there is no history the currentLEftTable will be underfiend 
            //if its underfined its underfiend then its the orignal state and therefore should be queue
            typeOfData = currentLeftTable.type || EndpointTypes.QUEUE;
        }
        switch (typeOfData) {
            case EndpointTypes.QUEUE:
                return (
                    <QueueList
                        queueData={this.state.queueData}
                        clickFunction={this.handleQueueRowClick}
                        currentlySelectedName={currentLeftTable.name}
                    />
                );
            case EndpointTypes.TOPIC:
                return (
                    <TopicList
                        topicData={this.state.topicData}
                        clickFunction={this.handleTopicRowClick}
                        currentlySelectedName={currentLeftTable.name}
                    />
                );

            case EndpointTypes.SUBSCRIPTION:
                return (
                    <SubscriptionList
                        subscriptionData={this.state.subscriptionData}
                        clickFunction={this.handleSubscriptionRowClick}
                        currentlySelectedName={currentLeftTable.name}
                    />
                );
            case EndpointTypes.MESSAGE:
                return (
                    <MessageList
                        messageData={this.state.messageData}
                    />
                );
            default:
                throw new Error('Invalid endpoint type.');
        }
    }

    HandleBreadCrumbClick = (type, newLength) => {
        switch (type) {
            case (EndpointTypes.QUEUE):
                this.breadCrumbHistory = this.breadCrumbHistory.slice(0, newLength + 1);
                this.updateAllQueueData();
                this.updateEndpointMessageData();
                this.setState({
                    rightTableType: EndpointTypes.MESSAGE
                });
                break;
            case (EndpointTypes.TOPIC):
                this.breadCrumbHistory = this.breadCrumbHistory.slice(0, newLength + 1);
                this.updateAllTopicData();
                this.updateTopicSubscriptionData();
                this.setState({
                    rightTableType: EndpointTypes.SUBSCRIPTION
                });
                break;
            case (EndpointTypes.SUBSCRIPTION):
                this.breadCrumbHistory = this.breadCrumbHistory.slice(0, newLength + 1);
                this.updateTopicSubscriptionData();
                this.updateEndpointMessageData();
                this.setState({
                    rightTableType: EndpointTypes.MESSAGE
                });
                break;
            default:
                this.breadCrumbHistory = this.breadCrumbHistory.slice(0, newLength + 1);
                this.resetInitialStateData();
                break;
        }

    }


    getBreadcrumbElement = () => {
        const breadcrumbStyle = css`
            float: left;
            height:2px;
            margin:2px;
        `;

        const breadcrumbItems = [];
        for (let i = 0; i < this.breadCrumbHistory.length; i++) {
            if (i === this.breadCrumbHistory.length - 1) {
                breadcrumbItems.push(
                    <Breadcrumb.Item onClick={() => this.HandleBreadCrumbClick(this.breadCrumbHistory[i].type, i)} active >
                        {this.breadCrumbHistory[i].name}
                    </Breadcrumb.Item>
                );
            } else {
                breadcrumbItems.push(
                    <Breadcrumb.Item onClick={() => this.HandleBreadCrumbClick(this.breadCrumbHistory[i].type, i)} >
                        {this.breadCrumbHistory[i].name}
                    </Breadcrumb.Item>
                );
            }
        }
        return (
            <Breadcrumb className={breadcrumbStyle} >
                {breadcrumbItems}
            </Breadcrumb>
        );
    }

    render() {
        //QQ change width when side buttons are removed
        const displayStyle = css`
            width: 30%;
            margin-left: 10px;
            margin-right:10px;
            display: inline-block; /*to allow tables to be displayed side by side*/
        `;
        const outerDivDisplay = css`
            display: block;
            `;

        const breadCrumbDisplay = css`
            display: block;
            height:40px;
            margin:2px;
            `;


        const leftBox = this.getList();
        const rightBox = this.getList(true);
        const leftType = this.breadCrumbHistory[this.breadCrumbHistory.length - 1].type || EndpointTypes.QUEUE;

        return (
            <div >
                <div className={breadCrumbDisplay} >
                    {this.getBreadcrumbElement()}
                </div>
                <div className={outerDivDisplay}>
                    <div className={displayStyle}>
                        <h2>{typeToTitle(leftType)}</h2>
                        {leftBox}
                    </div>
                    <div className={displayStyle}>
                        <h2>{typeToTitle(this.state.rightTableType)}</h2>
                        {rightBox}
                    </div>
                </div>
            </div>
        );
    }
}

