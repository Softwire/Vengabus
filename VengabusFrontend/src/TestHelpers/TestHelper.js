import { Button } from "react-bootstrap";

class TestHelper {
    /*
     * It returns a promise that waits until the state of the components is updated and then continues.
     * 
     * Expected usage is: 
     *     return testHelper.afterReactHasUpdated().then(() => {
     *         expect('actualValue').toEqual('expectedValue');
     *     });
     * 
     * Notes on multi-layered async tests (e.g. wait for this, and then wait for that, and then expect a result) will be forthcoming.
     */
    afterReactHasUpdated(noArgsGiven) {
        if (noArgsGiven) {
            throw new Error('afterReactHasUpdated() does not expect to take any arguments. See expected usage for further details.');
        }

        return new Promise(function (resolve) {
            setTimeout(resolve, 0);
        });
    }
    clickElementWithId(wrapper, id) {
        wrapper.find(id).at(0).simulate("click");
    }
}

export const testHelper = new TestHelper();
