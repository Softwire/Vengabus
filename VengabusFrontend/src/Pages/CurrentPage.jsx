import React, { Component } from 'react';
import { Alert } from 'react-bootstrap';
import { PAGES, pageSwitcher } from './PageSwitcherService';
import { HomePage } from './HomePage';
import { DemoPage } from './DemoPage';
import { TwoListDisplayPage } from './TwoListDisplayPage';
import { SendMessagePage } from './SendMessagePage';

export class CurrentPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            currentPage: pageSwitcher.currentPage,
            data: pageSwitcher.data
        };
    }

    componentDidMount() {
        pageSwitcher.registerForSwitchUpdates(this.switchToPage);
    }

    componentWillUnmount() {
        pageSwitcher.deregisterForSwitchUpdates(this.switchToPage);
    }

    switchToPage = (page, data) => {
        this.setState({
            currentPage: page,
            data: data
        });
    };

    onChildInput = (e, targetProperty) => {
        const mutation = {};
        mutation[e.target.name] = e.target.value;
        this.setState(mutation);
    };

    render() {
        let returnValue;
        switch (this.state.currentPage) {
            case PAGES.HomePage:
                returnValue = <HomePage />;
                break;

            case PAGES.DemoPage:
                returnValue = <DemoPage />;
                break;
            case PAGES.QueueDispayPage:
                returnValue = <TwoListDisplayPage />;
                break;

            case PAGES.SendMessagePage:
                const messageTemplate = this.state.data;
                returnValue = <SendMessagePage data={messageTemplate ? { ...messageTemplate } : undefined} />;
                break;

            default:
                returnValue =
                    <Alert bsStyle="danger">
                        <p>Requested Unknown Page: {this.state.currentPage}</p>
                    </Alert>
                    ;
        }
        return returnValue;
    }
}
