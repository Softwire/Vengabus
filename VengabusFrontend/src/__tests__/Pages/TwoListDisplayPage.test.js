import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';
import { TwoListDisplayPage } from "../../Pages/TwoListDisplayPage";
import { testHelper } from '../../TestHelpers/TestHelper';

//QQ DATA IS HERE 

jest.mock('../../AzureWrappers/VengaServiceBusService', () => ({
    VengaServiceBusService: class {
        constructor() {

        }
        listTopics = () => {
            return Promise.resolve({
                data: [
                    {
                        name: "test1",
                        subscriptionCount: 12,
                        topicStatus: "Active"
                    },
                    {
                        name: "test2",
                        subscriptionCount: 13,
                        topicStatus: "Active"
                    },
                    {
                        name: "test3",
                        subscriptionCount: 13,
                        topicStatus: "Active"
                    }
                ]
            });
        }


        listQueues = () => {
            return Promise.resolve({
                data: [
                    {
                        name: "test1",
                        activeMessageCount: 12,
                        deadletterMessageCount: 17
                    },
                    {
                        name: "test2",
                        activeMessageCount: 13,
                        deadletterMessageCount: 15
                    },
                    {
                        name: "test3",
                        activeMessageCount: 13,
                        deadletterMessageCount: 18
                    }
                ]
            });

        }

        listQueueMessages = () => {
            return Promise.resolve({
                data: [
                    {
                        predefinedProperties: { messageId: "test1" },
                        messageBody: "apple"

                    },
                    {
                        predefinedProperties: { messageId: "test2" },
                        messageBody: "banna"
                    },
                    {
                        predefinedProperties: { messageId: "test3" },
                        messageBody: "carrot"
                    }
                ]
            });

        }
        listSubscriptions = () => {
            return Promise.resolve({
                data: [
                    {
                        name: "test1",
                        activeMessageCount: 12,
                        deadletterMessageCount: 17
                    },
                    {
                        name: "test2",
                        activeMessageCount: 13,
                        deadletterMessageCount: 15
                    },
                    {
                        name: "test3",
                        activeMessageCount: 13,
                        deadletterMessageCount: 18
                    }
                ]
            });

        }

        listSubscriptionMessages = () => {
            return Promise.resolve({
                data: [
                    {
                        predefinedProperties: { messageId: "test1" },
                        messageBody: "apple"

                    },
                    {
                        predefinedProperties: { messageId: "test2" },
                        messageBody: "banna"
                    },
                    {
                        predefinedProperties: { messageId: "test3" },
                        messageBody: "carrot"
                    }
                ]
            });
        }
    }
}));


it('renders correctly before data is added', () => {
    let messagePropertyInput = renderer.create(
        <TwoListDisplayPage />);
    expect(messagePropertyInput.toJSON()).toMatchSnapshot();
});


it('renders queues and topic titles', () => {
    let wrapper = mount(<TwoListDisplayPage />);

    const leftTitle = wrapper.find('#leftTitle').text();
    const rightTitle = wrapper.find('#rightTitle').text();
    expect(leftTitle).toBe("Queues");
    expect(rightTitle).toBe("Topics");
});


it('queues and topics populate', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();

        const queueList = wrapper.find('#QueueTable').find("#Data");
        const topicList = wrapper.find('#TopicTable').find("#Data");
        expect(queueList.exists()).toBe(true);
        expect(topicList.exists()).toBe(true);
    }
    );

});


it('clicking Queues retrieves messages', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    const queueList = wrapper.find('#QueueTable');
    queueList.props().clickFunction('e', { name: "test1" });

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();
        const messageList = wrapper.find('#MessageTable');
        const rightTitle = wrapper.find('#rightTitle').text();
        expect(rightTitle).toBe("Messages");
        expect(messageList.exists()).toBe(true);

    });
});

it('clicking Topics retrieves subscriptions', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    const topic = wrapper.find('#TopicTable');
    topic.props().clickFunction('e', { name: "test1" });

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();

        const subscriptionTable = wrapper.find('#SubscriptionTable');
        const rightTitle = wrapper.find('#rightTitle').text();

        expect(rightTitle).toBe("Subscriptions");
        expect(subscriptionTable.exists()).toBe(true);

    });
});


it('clicking Subscriptions retrieves messages', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    const topicTable = wrapper.find('#TopicTable');
    topicTable.props().clickFunction('e', { name: "test1" });

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();

        const subscriptionTable = wrapper.find('#SubscriptionTable');
        subscriptionTable.props().clickFunction('e', { name: "test1" });

        return testHelper.afterReactHasUpdated().then(() => {
            wrapper.update();

            const messageTable = wrapper.find('#MessageTable');
            const rightTitle = wrapper.find('#rightTitle').text();

            expect(rightTitle).toBe("Messages");
            expect(messageTable.exists()).toBe(true);

        });
    });
});


it('Home breadCrumb resets state', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    const queueTable = wrapper.find('#QueueTable');
    queueTable.props().clickFunction('e', { name: "test1" });

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();

        const homeBreadCrumb = wrapper.find('#Home').hostNodes();
        homeBreadCrumb.simulate("click");

        return testHelper.afterReactHasUpdated().then(() => {
            wrapper.update();

            const queueTable = wrapper.find('#QueueTable').find("#Data");
            const topicTable = wrapper.find('#TopicTable').find("#Data");

            const rightTitle = wrapper.find('#rightTitle').text();
            expect(rightTitle).toBe("Topics");

            const leftTitle = wrapper.find('#leftTitle').text();
            expect(leftTitle).toBe("Queues");

            expect(queueTable.exists()).toBe(true);
            expect(topicTable.exists()).toBe(true);
        });
    });
});


it('Can go back to Topic from Subscription using BreadCrumbs', () => {
    let wrapper = mount(<TwoListDisplayPage />);
    wrapper.instance().resetInitialStateData();

    const topicTable = wrapper.find('#TopicTable');
    topicTable.props().clickFunction('e', { name: "test1" });

    return testHelper.afterReactHasUpdated().then(() => {
        wrapper.update();

        const subscriptionTable = wrapper.find('#SubscriptionTable');
        subscriptionTable.props().clickFunction('e', { name: "test2" });

        return testHelper.afterReactHasUpdated().then(() => {
            wrapper.update();

            const topicBreadCrumb = wrapper.find('#test1').hostNodes();
            topicBreadCrumb.simulate("click");

            return testHelper.afterReactHasUpdated().then(() => {
                wrapper.update();

                const subscriptionTable = wrapper.find('#SubscriptionTable');
                const topicTable = wrapper.find('#TopicTable');

                const rightTitle = wrapper.find('#rightTitle').text();
                const leftTitle = wrapper.find('#leftTitle').text();

                expect(rightTitle).toBe("Subscriptions");
                expect(leftTitle).toBe("Topics");

                expect(subscriptionTable.exists()).toBe(true);
                expect(topicTable.exists()).toBe(true);
            });
        });
    });
});