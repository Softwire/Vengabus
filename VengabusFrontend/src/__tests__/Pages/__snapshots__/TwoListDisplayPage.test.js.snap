// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly before data is added 1`] = `
.emotion-2 {
  width: 40%;
  margin-left: 10px;
  margin-right: 10px;
  display: inline-block;
}

.emotion-4 {
  display: block;
}

.emotion-1 {
  display: block;
  height: 40px;
  margin: 2px;
}

.emotion-0 {
  float: left;
  height: 2px;
  margin: 2px;
}

<div>
  <div
    className="emotion-1"
  >
    <ol
      aria-label="breadcrumbs"
      className="emotion-0 breadcrumb"
      role="navigation"
    >
      <li
        className="active"
      >
        <span
          id="Home"
          onClick={[Function]}
        >
          Home
        </span>
      </li>
    </ol>
  </div>
  <div
    className="emotion-4"
  >
    <div
      className="emotion-2"
    >
      <h2
        id="leftTitle"
      >
        Queues
      </h2>
      <p
        id="warning"
      >
        No data has been retrieved yet.
      </p>
    </div>
    <div
      className="emotion-2"
    >
      <h2
        id="rightTitle"
      >
        Topics
      </h2>
      <p
        id="warning"
      >
        No data has been retrieved yet.
      </p>
    </div>
  </div>
</div>
`;
